app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -78,72 +76,51 @@ class TransferSendManager @Inject constructor(
+            }
+            if (exception is IntegrityException) {
+                _sendStatus.update { SendStatus.Refused }
+            } else {
+                _sendStatus.update { SendStatus.Failure }
+            }
+        }
     }
 
     //region App Integrity
-    private suspend inline fun withIntegrityToken(onSuccess: (attestationToken: String) -> Unit, onRefused: () -> Unit = {}) {
-        var attestationToken: String? = null
-
-        coroutineScope {
-            appIntegrityManager.getChallenge(
-                onSuccess = { launch { attestationToken = requestAppIntegrityToken(appIntegrityManager) } },
-                onFailure = {},
-            )
-        }
-
-        attestationToken?.let(onSuccess) ?: onRefused()
+    /** Handle [IntegrityException] and all the different exceptions that can be thrown */
+    private suspend inline fun getAttestationToken(): String {
+        val challenge = appIntegrityManager.getChallenge()
+        return requestAppIntegrityToken(challenge)
     }
 
-    private suspend fun requestAppIntegrityToken(appIntegrityManager: AppIntegrityManager): String? {
-        var attestationToken: String? = null
-
-        coroutineScope {
-            appIntegrityManager.requestClassicIntegrityVerdictToken(
-                onSuccess = { token ->
-                    SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful")
-                    launch { attestationToken = getApiIntegrityVerdict(appIntegrityManager, token) }
-                },
-                onFailure = {},
-            )
-        }
-
-        return attestationToken
+    private suspend fun requestAppIntegrityToken(challenge: String): String {
+        val token = appIntegrityManager.requestClassicIntegrityVerdictToken(challenge)
+        SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful")
+        return getApiIntegrityVerdict(appIntegrityManager, token)
     }
 
-    private suspend fun getApiIntegrityVerdict(appIntegrityManager: AppIntegrityManager, appIntegrityToken: String): String? {
-        var token: String? = null
-
-        appIntegrityManager.getApiIntegrityVerdict(
+    private suspend fun getApiIntegrityVerdict(appIntegrityManager: AppIntegrityManager, appIntegrityToken: String): String {
+        val token = appIntegrityManager.getApiIntegrityVerdict(
             integrityToken = appIntegrityToken,
             packageName = BuildConfig.APPLICATION_ID,
             targetUrl = sharedApiUrlCreator.createUploadContainerUrl,
-            onSuccess = { attestationToken ->
-                SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "API verdict check")
-                Log.i(APP_INTEGRITY_MANAGER_TAG, "getApiIntegrityVerdict: $attestationToken")
-                token = attestationToken
-            },
-            onFailure = {},
         )
 
+        SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "API verdict check $token")
```

Comment:
I'd rather not send the token to sentry, that's why I let a Simple `Log.i`for us to have the information, when testing
Plus it's useless to have it on Sentry as it only lasts 5 min